### ISSUE TITLE
Optional BigDecimal doesn't compile

### ISSUE THREAD

**Issue by Miro-Marinov** [julianpeeters/sbt-avrohugger#69]
Hi, I got an error when attempting to specify a field to be an optional decimal one (following the example [here](https://github.com/julianpeeters/sbt-avrohugger/blob/master/src/sbt-test/avrohugger/GenericSerializationTests/src/main/avro/logical.avdl#L6)). Relevant custom type config is like so:

```
avroScalaSpecificCustomTypes in Compile := {
  avrohugger.format.SpecificRecord.defaultTypes.copy(
    enum = avrohugger.types.JavaEnum,
    uuid = avrohugger.types.JavaUuid,
    decimal = avrohugger.types.ScalaBigDecimal,
    timestampMillis = avrohugger.types.JavaTimeInstant,
    // Multiple records within a protocol definition will be placed in the same file inheriting from a base sealed trait
    protocol = ScalaADT
  )
}
```

Avdl record:

```
@namespace("payments")
protocol Payment {

  record Payment {
    string id;
    timestamp_ms  createdAt;
    union {decimal(16, 2), null} amount;
  }
}
```

Plugin version:
`addSbtPlugin("com.julianpeeters" % "sbt-avrohugger" % "2.0.0-RC15")`

The error that I get when I do `sbt ";clean;compile"` is:

```
[error] D:\domain\target\scala-2.12\src_managed\main\compiled_avro\payments\Payment.scala:24:21: value decimalConversion is not a member of object payments.Payment
[error]             Payment.decimalConversion.toBytes(bigDecimal, schema, decimalType)
[error]                     ^
[error] D:\domain\scala-2.12\src_managed\main\compiled_avro\payments\Payment.scala:51:34: value decimalConversion is not a member of object payments.Payment
[error]               BigDecimal(Payment.decimalConversion.fromBytes(buffer, schema, decimalType))
[error]                                  ^
[error] two errors found
[error] (Compile / compileIncremental) Compilation failed
```

If I replace the `union {decimal(16, 2), null}` with just `decimal(16, 2)` it works. When I try `decimal = avrohugger.types.ScalaBigDecimalWithPrecision,` instead of `decimal = avrohugger.types.ScalaBigDecimal`, with or without the union I get an error similar to:

```
[error] D:\domain\target\scala-2.12\src_managed\main\compiled_avro\payments\Payment.scala:6:82: not found: type @@
[error] case class Payment(var id: String, var createdAt: java.time.Instant, var amount: @@[scala.math.BigDecimal, ((shapeless.Nat._1, shapeless.Nat._6), shapeless.Nat._2)]) extends
org.apache.avro.specific.SpecificRecordBase {
[error]                                                                                  ^
[error] D:\domain\target\scala-2.12\src_managed\main\compiled_avro\payments\Payment.scala:47:22: not found: type @@
[error]       }.asInstanceOf[@@[scala.math.BigDecimal, ((shapeless.Nat._1, shapeless.Nat._6), shapeless.Nat._2)]]
[error]                      ^
[error] 8 errors found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 8 s, completed Jan 19, 2019 6:58:57 PM
```

What am I doing wrong? Thanks in advance!

**Comment by julianpeeters** [comment]
Thanks for the details, @miro-marinov, that looks like a duplicate of https://github.com/julianpeeters/avrohugger/issues/107

Looks like the fix stalled, I'll see if I can't polish that off with the classpath fix

**Comment by julianpeeters** [comment]
Alrighty! thanks for the patience. If it's still of interest, this is expected to now be fixed in sbt-avrohugger 2.0.0-RC16
